### PR TITLE
Deduplicate FormatStringID symbols

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This caused many surprises.

### CODE REVIEW

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 1]
```diff
@@ -460,7 +460,7 @@ static u8string OpenSystemFileBrowser(bool isSave)
             if (buffer.empty())
             {
                 // Use localised "Unnamed Park" if park name was empty.
-                buffer = FormatStringID(STR_UNNAMED_PARK, nullptr);
+                buffer = FormatStringIDLegacy(STR_UNNAMED_PARK, nullptr);
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/LoadSave.cpp:463]: Anything that is calling this with a nullptr is probably incorrect it should probably be calling `LanguageGetString`

**Comment by AaronVanGeffen** [comment]
Could you elaborate how this deduplicates symbols? I suppose renaming this particular `FormatStringID` function disambiguates from another `FormatStringID`, rather than relying on signature overloading?

**Comment by duncanspumpkin** [comment]
> Could you elaborate how this deduplicates symbols? I suppose renaming this particular `FormatStringID` function disambiguates from another `FormatStringID`, rather than relying on signature overloading?

There was no signature overloading so it was just a simple collision if you included both headers. One was in the openrct2 namespace the other was not so you could use that to select them but it was v. horrible.

**Review by tupaschoal — APPROVED** [review]

**Comment by Gymnasiast** [comment]
If I understand correctly, this renames one of the FormatStringID() functions to FormatStringIDLegacy() in order to avoid collisions. Is that correct?

File: src/openrct2-ui/windows/LoadSave.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2-ui/windows/LoadSave.cpp:463]: I’m already on this.

**Comment by duncanspumpkin** [comment]
> If I understand correctly, this renames one of the FormatStringID() functions to FormatStringIDLegacy() in order to avoid collisions. Is that correct?

Yup

**Comment by Gymnasiast** [comment]
Ok, that makes sense.

I’d like to have https://github.com/OpenRCT2/OpenRCT2/pull/21571 merged first and then this one rebased.

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**